game/world/managers/objects/units/player/PlayerManager.py

```diff
@@ -1241,6 +1241,13 @@ def generate_rage(self, damage_info, is_player=False):
                                                            self.has_form(ShapeshiftForms.SHAPESHIFT_FORM_BEAR)):
             self.set_rage(self.power_2 + Formulas.PlayerFormulas.calculate_rage_regen(damage_info, is_player=is_player))
 
+    def generate_rage_on_received_damage(self, damage_info):
+        # Warriors or Druids in Bear form
+        if self.player.class_ == Classes.CLASS_WARRIOR or (self.player.class_ == Classes.CLASS_DRUID and
```

Comment:
Warriors can only regen rage when they receive damage if they are in Defensive or Battle stances.

Reply:
it's done